src/ResourceManager/PowerBIEmbedded/Commands.PowerBI/Commands/ResumeAzureRmPowerBIEmbeddedCapacity.cs

```diff
@@ -0,0 +1,111 @@
+        protected const string ObjectParameterSet = "ByInputObject";
+        protected const string ResourceIdParameterSet = "ByResourceId";
+
+        [Parameter(
+            ParameterSetName = CmdletParametersSet,
+            Mandatory = true,
+            Position = 0,
+            HelpMessage = "Name of a specific capacity.")]
+        [ValidateNotNullOrEmpty]
+        public string Name { get; set; }
+
+        [Parameter(
+            ParameterSetName = CmdletParametersSet,
+            Mandatory = false,
+            HelpMessage = "Name of resource group under which to retrieve the capacity.")]
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupName { get; set; }
+
+        [Parameter(
+            ParameterSetName = ResourceIdParameterSet,
+            Mandatory = true,
+            Position = 0,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "PowerBI Embedded Capacity ResourceID.")]
+        [ValidateNotNullOrEmpty]
+        public string ResourceId { get; set; }
+
+        [Parameter(
+            ParameterSetName = ObjectParameterSet,
+            Mandatory = true,
+            Position = 0,
+            ValueFromPipeline = true,
+            HelpMessage = "PowerBI Embedded Capacity object.")]
+        [ValidateNotNullOrEmpty]
+        public PSPowerBIEmbeddedCapacity InputObject { get; set; }
+
+        [Parameter(Mandatory = false)]
+        public SwitchParameter PassThru { get; set; }
+
+        public override void ExecuteCmdlet()
+        {
+            string capacityName = Name;
+            string resourceGroupName = ResourceGroupName;
+
+            if (!string.IsNullOrEmpty(this.ResourceId))
+            {
+                PowerBIUtils.GetResourceGroupNameAndCapacityName(ResourceId, out resourceGroupName, out capacityName);
+            }
+            else if (InputObject != null)
+            {
+                PowerBIUtils.GetResourceGroupNameAndCapacityName(InputObject.Id, out resourceGroupName, out capacityName);
+            }
+
+            if (string.IsNullOrEmpty(capacityName))
+            {
+                WriteExceptionError(new PSArgumentNullException("Name", "Name of capacity not specified"));
+            }
+
+            if (ShouldProcess(capacityName, Resources.ResumingPowerBIEmbeddedCapacity))
+            {
+                PSPowerBIEmbeddedCapacity capacity = null;
+                if (!PowerBIClient.TestCapacity(resourceGroupName, capacityName, out capacity))
+                {
+                    throw new InvalidOperationException(string.Format(Properties.Resources.CapacityDoesNotExist, capacityName));
+                }
+
+                PowerBIClient.ResumeCapacity(resourceGroupName, capacityName);
+
+                if (PassThru.IsPresent)
+                {
+                    // Update the capacity current state 
+                    if (!PowerBIClient.TestCapacity(resourceGroupName, capacityName, out capacity))
```

Comment:
@selasagi why is this second `TestCapacity` call being made when `PassThru` is provided? Shouldn't it be made regardless of whether or not the user wants output?

Reply:
If output required, I want that the output capacity state will be updateed/synced